app/src/main/java/com/example/una/ImpactFragment.java

```diff
@@ -18,5 +30,24 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
     @Override
     public void onViewCreated(View view, Bundle savedInstanceState) {
         super.onViewCreated(view, savedInstanceState);
+
+        signOutButton = view.findViewById(R.id.signOutButton);
+
+        signOutButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                AuthUI.getInstance()
+                        .signOut(getContext())
+                        .addOnCompleteListener(new OnCompleteListener<Void>() {
+                            @Override
+                            public void onComplete(@NonNull Task<Void> task) {
+                                Intent signOutIntent = new Intent(getContext(), LoginActivity.class);
+                                startActivity(signOutIntent);
+                                // getActivity().finish();
```

Comment:
no need for commented code

Reply:
you might want to finish this activity too. Once the user has signed out, they shouldn't be able to access `MainActivity` unless they sign in again